### ISSUE TITLE
MaxReconnectToBroker is useless for consumer

### ISSUE THREAD

**Issue by omnilight** [apache/pulsar-client-go#496]
#### Expected behavior

When we set `ConsumerOptions.MaxReconnectToBroker` to some value other than infinite, then when reconnection count reaches this value consumer should return with an error.

#### Actual behavior

Consumer just stuck, because [reconnectToBroker](https://github.com/apache/pulsar-client-go/blob/master/pulsar/consumer_partition.go#L861) just exists leaving consumer useless

#### Steps to reproduce

1. Set option to `ConsumerOptions.MaxReconnectToBroker = 2`
2. Begin consuming messages
3. Do something with broker (say, stop it)
4. Messages would stop consuming
5. Start broker
6. Messages won't appear because consumer would not reconnect

#### System configuration
**Pulsar version**: 2.7

**Comment by omnilight** [comment]
Added pull request with fix

**Comment by jonyhy96** [comment]
I think this is what `MaxReconnectToBroker` means, it won't try to reconnect to broker after given `MaxReconnectToBroker` tries. 😄 
By the way, we do need to close useless consumer when we are not trying to reconnect to broker.

**Comment by omnilight** [comment]
Yes, but in the case it stops reconnecting consumer will just stuck and no notification would be passed to the caller. In the pr I've added closing consumer in this case. For me that seems valid behaviour

**Comment by omnilight** [comment]
@wolfstudy Hi! Sorry for direct mention, but could we proceed with this issue in any way?

**Comment by wolfstudy** [comment]
> @wolfstudy Hi! Sorry for direct mention, but could we proceed with this issue in any way?

Thanks @omnilight work for this, agree with you.

**Comment by barbashov** [comment]
Are there any plans for this PR to be included in a release at some point? We're suffering a little bit from this bug :)

@wolfstudy friendly ping

**Comment by barbashov** [comment]
@wolfstudy any updates on that please? We struggle with that behavior
